tests/config/test_search.py

```diff
@@ -9,119 +9,170 @@
+        check = CheckAvailMock(expected)
+        side_effects = check.check
+        aval_mock.side_effect = side_effects
+        assert list(search.get_config_files()) == expected
 
 
 def test_default_returned_slot_3():
     """Test config returned from ./config"""
     with patch('atkinson.config.search._check_available') as aval_mock:
-        aval_mock.side_effect = [False, False, True]
-        assert search.get_config_file() == os.path.realpath('./configs/config.yml')
+        expected = [os.path.expanduser('~/.atkinson/config.yml')]
+        check = CheckAvailMock(expected)
+        side_effects = check.check
+        aval_mock.side_effect = side_effects
+        assert list(search.get_config_files()) == expected
 
 
 def test_default_not_found():
     """Test config can't be found in the default paths"""
     with patch('atkinson.config.search._check_available') as aval_mock:
-        aval_mock.side_effect = [False, False, False]
-        assert search.get_config_file() == ''
+        aval_mock.return_value = False
+        assert list(search.get_config_files()) == []
 
 
 def test_custom_returned_slot_1():
     """Test config returned from ~/.atkinson"""
     with patch('atkinson.config.search._check_available') as aval_mock:
-        aval_mock.side_effect = [True]
-        expected = os.path.expanduser('~/.atkinson/my_config.yml')
-        assert search.get_config_file(filename='my_config.yml') == expected
+        file_name = 'my_config.yml'
+        expected = [os.path.join(os.path.expanduser('~/.atkinson'), file_name)]
+        check = CheckAvailMock(expected)
+        aval_mock.side_effect = check.check
+        assert list(search.get_config_files(filenames=file_name)) == expected
 
 
 def test_custom_returned_slot_2():
     """Test config returned from /etc/atkinson"""
     with patch('atkinson.config.search._check_available') as aval_mock:
-        aval_mock.side_effect = [False, True]
-        assert search.get_config_file(filename='my_config.yml') == '/etc/atkinson/my_config.yml'
+        file_name = 'my_config.yml'
+        expected = [os.path.join('/etc/atkinson', file_name)]
+        check = CheckAvailMock(expected)
+        aval_mock.side_effect = check.check
+        assert list(search.get_config_files(filenames=file_name)) == expected
 
 
 def test_custom_returned_slot_3():
-    """Test config returned from ./config"""
+    """Test config returned from ./configs"""
     with patch('atkinson.config.search._check_available') as aval_mock:
-        aval_mock.side_effect = [False, False, True]
-        expected = os.path.realpath('./configs/my_config.yml')
-        assert search.get_config_file(filename='my_config.yml') == expected
+        file_name = 'my_config.yml'
+        expected = [os.path.join(os.path.realpath('./configs'), file_name)]
+        check = CheckAvailMock(expected)
+        aval_mock.side_effect = check.check
+        assert list(search.get_config_files(filenames=file_name)) == expected
 
```

Comment:
Above 3 could probably also collapse to 1 with parametrize.

Reply:
Updated